experimental/Schemes/CoherentSheaves.jl

```diff
@@ -1193,10 +1546,90 @@ end
   return matrix(map(presentation(M), 1))
 end
 
-function fill_with_lazy_glueings!(C::Covering, X::AbsCoveredScheme)
-  # TODO: Fill in all the implicit and restricted glueings from X.
-  # We assume that all patches in C are in a tree structure eventually 
-  # leading to affine charts of X.
-  return C
+########################################################################
+# Projectivization of vector bundles                                   #
+#
+# To any vector bundle E → X one can associate its projectivization 
+# ℙ(E) → X. In general, E will be given as a locally free 
+# AbsCoherentSheaf.
+########################################################################
+
+function projectivization(E::AbsCoherentSheaf; 
+    var_names::Vector{String}=Vector{String}(),
+    check::Bool=true
+  )
+  X = scheme(E)
+  check && (is_locally_free(E) || error("coherent sheaf must be locally free"))
+  C = trivializing_covering(E)
+  algebras = IdDict{AbsSpec, Union{MPolyQuo, MPolyRing}}()
+  on_patches = IdDict{AbsSpec, AbsProjectiveScheme}()
+  for U in patches(C)
+    F = E(U)
+    F isa FreeMod || error("modules must locally be free")
+    r = rank(F)
+    if length(var_names) == 0
+      var_names = ["s$i" for i in 0:r-1]
+    end
+    length(var_names) == r || error("number of names for the variables does not coincide with the local rank of the module")
+    RU = rees_algebra(E(U), var_names=var_names)
+    algebras[U] = RU
+    SU, _ = grade(RU)
+    PU = ProjectiveScheme(SU)
+    set_base_scheme!(PU, U)
+    on_patches[U] = PU
+  end
+  projective_glueings = IdDict{Tuple{AbsSpec, AbsSpec}, ProjectiveGlueing}()
+  OX = StructureSheafOfRings(X)
+
+  # prepare for the projective glueings
+  for (U, V) in keys(glueings(C))
+    P = on_patches[U]
+    SP = ambient_coordinate_ring(P)
+    Q = on_patches[V]
+    SQ = ambient_coordinate_ring(Q)
+    G = C[U, V]
+    UV, VU = glueing_domains(G)
+    f, g = glueing_morphisms(G)
+
+    PUV, PUVtoP = fiber_product(OX(U, UV), P)
+    QVU, QVUtoQ = fiber_product(OX(V, VU), Q)
+
+    # to construct the identifications of PUV with QVU we need to 
+    # express the generators of I(U) in terms of the generators of I(V)
+    # on the overlap U ∩ V. 
+    !(G isa Glueing) || error("method not implemented for this type of glueing")
```

Comment:
I(U) and I(V) are the modulus of OO(X) on the respective patches, correct?

Reply:
See comment above.